java/flight/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -58,14 +59,19 @@
 
   private final Location location;
   private final Server server;
+  // The executor used by the gRPC server. We don't use it here, but we do need to clean it up with the server.
+  // May be null, if a user-supplied executor was provided (as we do not want to clean that up)
+  @VisibleForTesting
+  final ExecutorService grpcExecutor;
```

Comment:
I think you could avoid exposing this if you added this to awaitTermination as well (but that is a little tricky to do correctly).  I'm OK if you don't want to do it.

Reply:
Hmm, how would that work in the test? We'd shut down the executor in awaitTermination and wait?

Reply:
I would have thought we shutdown the executorservice in shutdown

Reply:
Er, right. I can do that, but I still don't see how that'd let us avoid exposing it in test.

Reply:
i think it tests the behavior implicitly vs explicitly as the way you have it now.  Like I said I'm fine with either approach.